### DISCUSSION TITLE
Origin Trial - Topics API - All Users get "TrialNotAllowed"

### DISCUSSION THREAD

**Discussion by simonlCadent** [privacysandbox/privacy-sandbox-dev-support#37] [Replacing cross-site tracking]
Hi, 

I am attempting to run a test of the Topics API and we have 10-15 users on Chrome Beta v103.0.5060.42 with `chrome://flags/#privacy-sandbox-ads-apis` enabled. 

We have set up a page on our domain to check if the Topics API is supported and to try and log that browsers topics. Every user has gotten 'TrialNotAllowed' messages inside the DevTools under the Application tab while our token is returning Success. 

![image](https://user-images.githubusercontent.com/107489227/173608051-14bfd640-9900-4e9f-a252-a4a64117c7e2.png)

`document.browserTopics()` is coming back as being supported when checked via JS but if we try to log it, we get an empty array. 

![image](https://user-images.githubusercontent.com/107489227/173608522-660dc9ae-6af7-41ee-8d6a-37fa1f728acb.png)

Does this TrialNotAllowed message mean that the users browser has not been selected to be part of the trial and this is why we're not getting any results when trying to log the topics? It was my understanding that by enabling the flag the user would bypass this.

**Comment by samdutton** [comment]
Hi @sladt — thanks for the heads-up on this.

I think there are a couple of things going on here.

You don't need to enable `chrome://flags/#privacy-sandbox-ads-apis` for users on pages that provide a valid token. That's what's leading to the DevTools status. (I've just been talking to the engineers working on this, about how to make the status clearer, i.e. not just have `TrialNotAllowed` in DevTools in this situation. I've also suggested that the Enabled/Disabled setting in this case might be better as Enabled/Disabled/Default — or similar.) 

Topics is now at 50% of Beta users on pages that provide a valid token, but it will take some time for users to ramp up in the experiment and, once they're in the experiment it takes [at least a week](https://developer.chrome.com/docs/privacy-sandbox/topics/#how-does-the-topics-api-decide-which-callers-can-see-which-topic) to return topics. That's probably why you're getting an empty array from `document.browsingTopics()`.
